static/app/components/profiling/FrameStack/profileDetails.tsx

```diff
@@ -115,49 +144,57 @@ export function ProfileDetails(props: ProfileDetailsProps) {
 
       {detailsTab === 'device' ? (
         <DetailsContainer>
-          <DetailsRow>
-            {Object.entries(DEVICE_DETAILS_KEY).map(([label, key]) => {
-              const value = props.profileGroup.metadata[key];
-              return (
-                <DetailsRow key={key}>
-                  <span>
-                    <strong>{label}</strong>:
-                  </span>{' '}
-                  <span>
-                    {key === 'durationNS'
-                      ? nsFormatter(value)
-                      : value === undefined || value === ''
-                      ? t('ø')
-                      : value}
-                  </span>
-                </DetailsRow>
-              );
-            })}
-          </DetailsRow>
+          {Object.entries(DEVICE_DETAILS_KEY).map(([label, key]) => {
+            const value = props.profileGroup.metadata[key];
+            return (
+              <DetailsRow key={key}>
+                <strong>{label}:</strong>
+                <span>{renderValue(key, value, props.profileGroup)}</span>
+              </DetailsRow>
+            );
+          })}
         </DetailsContainer>
       ) : (
         <DetailsContainer>
           {Object.entries(PROFILE_DETAILS_KEY).map(([label, key]) => {
             const value = props.profileGroup.metadata[key];
 
+            if (key === 'organizationID') {
+              const org = organizations.find(o => o.id === String(value));
```

Comment:
@NisanthanNanthakumar I quickly glanced at it, but looks like that only gives you the current org the user is on. I guess it's impossible to see a different org anyways since the data is queries using the org param, but I'd still rather resolved it from all of the orgs the user has access to